portal/models/identifier.py

```diff
@@ -48,6 +48,9 @@ def from_fhir(cls, data):
     def __str__(self):
         return 'Identifier {0.use} {0.system} {0.value}'.format(self)
 
+    def __hash__(self):
```

Comment:
Need to read more about this change before I can comment ([this seems like a good reference](https://hynek.me/articles/hashes-and-equality/))

Reply:
@pbugni Thoughts?

Reply:
Did some reading and we should only need that method defined if we use `Identifier` objects in dicts and sets (we likely do).

It's unlikely the `system` and `value` will ever change, but I think you might want to prefer the primary key (a suggestion in the above article), if available, for a attribute that's less likely to change. eg:

```
return hash(self.id) if self.id else hash(self.system)*hash(self.value)
```

Reply:
Yes, I was getting an error because Identifiers are used in sets somewhere. Ok, I'll change to that.

Reply:
I like this suggested approach - let's add a `__key__` and use it in both equality and hash:  https://stackoverflow.com/a/2909119